### ISSUE TITLE
messaging.onMessage callback fires when re-opening an app on Android

### ISSUE THREAD

**Issue by dannycochran** [invertase/react-native-firebase#842]
### Issue

The messaging.onMessage callback fires when re-opening an app on Android.

For example with this code, open an app, then background the app (don't close it):

```javascript
// This gets called once on app mount.
firebaseApp.messaging().onMessage(notification => {
  // onMessage handler stuff, gets fired whenever the app is backgrounded and then re-opened.
});
```

Bring the app back up, and the callback for "onMessage" is fired.

### Environment

Environment:
  OS: macOS Sierra 10.12.6
  Node: 6.10.3
  Yarn: 1.3.2
  npm: 3.10.10
  Watchman: Not Found
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: Not Found

Packages: (wanted => installed)
  react: 16.2.0 => 16.2.0
  react-native: 0.52.2 => 0.52.2
  react-native-firebase: 3.2.5 => 3.2.5

**Comment by chrisbianca** [comment]
@dannycochran Thanks for raising - this will be addressed as part of the work on #595 which is now well underway

**Comment by chrisbianca** [comment]
Good news, the long awaited alpha of our messaging and notifications overhaul is now available!!

Check out the release notes here: https://github.com/invertase/react-native-firebase/releases/tag/v4.0.0-alpha.1

If you have any comments and suggestions or want to report an issue, come find us on [Discord](https://discord.gg/C9aK28N)
